### PR TITLE
compare recovered signer with tx.from

### PR DESCRIPTION
**Pull request by i-m-aditya**

### CODE REVIEW

File: src/executor/mod.rs
Comment (lispc) [reply src/executor/mod.rs:108]: will you use tiny-keccak?

**Comment by lispc** [comment]
the PR brings in too many lines of codes...   Is there some libs we can use?

I read the eth_tx.recover_from implemention, and thinks it is not too "patch friendly".   Maybe you can visit similar features (recover addr from sig) from alloy-rs?  if needed, we could switch to alloy-rs for (some?) crates?

File: src/executor/mod.rs
Comment (roynalnaruto) [reply src/executor/mod.rs:56]: For reference, this is being called:
https://docs.rs/ethers-core/latest/ethers_core/types/struct.Signature.html#method.recover

**Review by roynalnaruto — CHANGES_REQUESTED** [review]
As I mentioned, we will need some refactoring before we merge this PR, especially related to error handling in SBV. Before that, you cannot really test the malicious case (incorrect signature), as your code would panic and this is something that we don't want.

File: testdata/mainnet_blocks/5224657u.json
Comment (roynalnaruto) [inline testdata/mainnet_blocks/5224657u.json:1]: Was this committed by mistake? Not sure what the `5224657u` is?

File: src/executor/block_errors.rs
Comment (roynalnaruto) [inline src/executor/block_errors.rs:5]: Based on the following review comment, we will need to propagate the `SignatureError` (from `ethers`) as well. So we may need something like:

```rust
#[error(transparent)]
SenderRecovery(#[from] ethers_core::types::SignatureError),
#[error("signature verification error: from={0}, recovered={}")]
SenderMismatch(Address, Address)
```

**Review by roynalnaruto — CHANGES_REQUESTED** [review]
A few more recommendations

File: testdata/mainnet_blocks/5224657u.json
Comment (i-m-aditya) [reply testdata/mainnet_blocks/5224657u.json:1]: Previous 5224657.json doesn't contain the codes, so for testing I used 5224657u.json.

File: src/executor/block_errors.rs
Comment (roynalnaruto) [reply src/executor/block_errors.rs:5]: But as I mentioned, its best to do this refactoring in a separate PR. I already have a WIP PR for this. So we can keep #19 on hold until we get the refactoring merged.

File: testdata/mainnet_blocks/5224657u.json
Comment (lispc) [reply testdata/mainnet_blocks/5224657u.json:1]: @lightsing will update the json files to latest format soon. Let merge his PR first.

File: src/executor/block_errors.rs
Comment (i-m-aditya) [reply src/executor/block_errors.rs:5]: @roynalnaruto can you share your WIP PR?

Comment (roynalnaruto) [reply src/executor/block_errors.rs:5]: #24

**Comment by roynalnaruto** [comment]
Depends on #24

**Comment by lispc** [comment]
Again you added a new json, is that needed?

File: testdata/mainnet_blocks/5224657u.json
Comment (lightsing) [reply testdata/mainnet_blocks/5224657u.json:1]: check

**Comment by i-m-aditya** [comment]
> Again you added a new json, is that needed?

Removed.

**Review by lispc — APPROVED** [review]

**Pull request merged**
